catalog/core/catalog-core-commons/src/main/java/ddf/util/XPathHelper.java

```diff
@@ -250,14 +258,37 @@ public synchronized Object evaluate(String xpathExpressionKey, QName returnType,
                 .setNamespaceContext(nsContext);
 
         XPathExpression compiledExpression = XPathCache.getCompiledExpression(xpathExpressionKey);
-
         Thread thread = Thread.currentThread();
         ClassLoader loader = thread.getContextClassLoader();
         thread.setContextClassLoader(this.getClass()
                 .getClassLoader());
 
+        DocumentBuilderFactory df =DocumentBuilderFactory.newInstance();
+        df.setNamespaceAware(true);
         try {
-            return compiledExpression.evaluate(document, returnType);
+            df.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
+            df.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+        } catch (ParserConfigurationException e) {
+            throw new XPathExpressionException(e);
```

Comment:
Why throw in this case but the other case we continue?  Also, why can't we reuse the existing DocumentBuilderFactory the was setup in the constructor?